source/nut-dw/usr/sbin/nut-notify

```diff
@@ -50,16 +50,18 @@ if [ "$SHUTDOWN" == "sec_timer" ]; then
 fi
 
 trap "exit 0" SIGTERM
-if [ "$NOTIFYTYPE" == "ONLINE" ]
+if [ "$NOTIFYTYPE" == "ONLINE" ] || [ "$NOTIFYTYPE" == "BYPASS" ]
 then
-        if [ "$(type -t "ONLINE")" == "function" ]; then
+        if [ "$NOTIFYTYPE" == "ONLINE" ] && [ "$(type -t "ONLINE")" == "function" ]; then
             ( ONLINE 2>&1 | logger -t "xnut-notify-hooks.sh" ) &
         fi
-        echo "[${NAME}] UPS is online, any planned shutdowns were cancelled." | tee >(wall) >(logger -t "nut-notify")
-        $NOTIFY -e "${EVENT}" -s "Notice ${SUBJECT} Online" -d "UPS is online, any planned shutdowns were cancelled." -i "normal"
 
-        # Cause all instances of this script to exit.
-        killall -s SIGTERM "$(basename "$0")"
+        # If script is running, cancel it (excluding this PID)
+        if pgrep -x "$(basename "$0")" | grep -v "$$" > /dev/null; then
+            echo "[${NAME}] UPS power restored, any planned shutdowns were cancelled." | tee >(wall) >(logger -t "nut-notify")
+            $NOTIFY -e "${EVENT}" -s "Notice ${SUBJECT} Online" -d "UPS power restored, any planned shutdowns were cancelled." -i "normal"
+            killall -s SIGTERM "$(basename "$0")"
+        fi
```

Comment:
I think there are some issues here, this would only fire if another script instance was also running. There are, however, situations when we want to be notified of an UPS being back `ONLINE` (`OL`) without another script instance running. For example, this can happen when the user has no shutdown criteria configured and the UPS goes `OL` -> `OB` -> `OL` or if the second instance of the script (that is checking the shutdown criteria) has exited prematurely (due to an error).  Such users would want to be notified of the UPS being back online from the battery state (`OB`), otherwise it'd raise concerns.

Moving on, `killall -s SIGTERM "$(basename "$0")"` will still kill the script itself, if no additional PID exclusion is done here. Generally though, I think this section is not really needed and possibly wasting precious time because at this point the script will end anyhow. There seems to be no real downside in also sending the current instance a `SIGTERM`, which is trapped with `exit 0` anyway, resulting in the script ending as it would also do otherwise.